CHANGELOG.md

```diff
@@ -3,6 +3,9 @@ All notable changes to this project will be documented in this file based on the
 
 ## [Unreleased](https://github.com/ruflin/Elastica/compare/3.2.0...HEAD)
 
+### Backward Compatibility Breaks
+- All classes that depend on \Elastica\Client by typehint have been updated to accept a ClientInterface. Any classes that extend Elastica classes that depend on this typehint will need to be updated.
```

Comment:
I think adding this as a 4.0 BC break is acceptable, it means you can swap in a php-elasticsearch wrapper that implements ClientInterface transparently.